gravitee-apim-definition/gravitee-apim-definition-model/src/main/java/io/gravitee/definition/model/PathMapping.java

```diff
@@ -26,7 +26,16 @@ public class PathMapping {
     private PathMapping() {}
 
     public static Pattern buildPattern(String mappingExpression) {
-        var regex = mappingExpression.replaceAll(":[^/]*", "[^/]*") + "/*";
+        if (mappingExpression == null || mappingExpression.isBlank()) return Pattern.compile(".*");
+
+        var regex =
+            mappingExpression
+                .replaceAll("([+?^$\\[\\]{}|])", "\\\\$1")
+                .replace("(", "\\(")
+                .replace(")", "\\)")
+                .replaceAll(":[^/]*", "[^/]*") +
+            "/*";
```

Comment:
I’m not sure to have all context but maybe it can be made with


```suggestion
        var regex = Patter.quote(mappingExpression);
```

just a suggestion sorry if it’s not relevant

Reply:
Thank you @michel-barret for the suggestion. As we discussed, I tested it and Pattern.quote works for literal paths, but our logic needs to handle parameters and wildcards too.